File: it.unibo.bls19Local/src/main/java/it/unibo/bls/devices/arduino/LedProxyArduino.java
package it.unibo.bls.devices.arduino;

import it.unibo.bls.devices.mock.LedMock;
import it.unibo.bls.interfaces.ILed;
import jssc.SerialPort;

public class LedProxyArduino extends LedMock {
    private JSSCSerialComm conn;

    public static ILed create(String portNum, int rate){
        return new LedProxyArduino(  portNum, rate );
    }
    public LedProxyArduino( String portNum, int rate ){
        configure( portNum, rate );
    }

    protected void configure( String portNum, int rate ){
        try {
            //conn = new JSSCSerialComm( rate );
            //conn.connect(portNum);
            conn = JSSCSerialComm.getSerialConn(portNum, rate);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
    @Override //LedMock
    public void turnOn(){
        super.turnOn();
        conn.writeData("1");
      }
    @Override //LedMock
    public void turnOff() {
        super.turnOff();
        conn.writeData("0");
    }
}

File: it.unibo.eclipse.qak.chainPlus/src/resources/ledProxyArduino.kt
package resources

import it.unibo.bls.devices.arduino.JSSCSerialComm


class ledProxyArduino( val  portNum : String,  val rate : Int) {
	companion object {
		lateinit var proxy : ledProxyArduino
		lateinit var conn  : JSSCSerialComm
		fun create( portNum: String,  rate: Int){
			proxy = ledProxyArduino(  portNum, rate )
		}
		
	}
	init{
		try {
			 conn = JSSCSerialComm.getSerialConn(portNum, rate)
		 } catch ( e : Exception ) {
            println("ERROR $e");
        }
	}
		fun turnOn(){
			conn.writeData("1");
		}
		fun turnOff(){
			conn.writeData("0");
		}
}